server/qrops.go

```diff
@@ -0,0 +1,273 @@
+	}
+	paddedImg := images.AddPadding(img, padding)
+	circle := images.CreateCircle(img)
+	centeredImg := images.PlaceCircleInCenter(paddedImg, circle)
+	resultBytes, err := images.EncodePNG(centeredImg)
+	if err != nil {
+		return nil, fmt.Errorf("encoding PNG failed: %v", err)
+	}
+	return resultBytes, nil
+}
+
+func GetLogoImage(multiaccountsDB *multiaccounts.Database, keyUID string, imageName string) ([]byte, error) {
+	var (
+		imageFolderBasePath = "../_assets/tests/"
+		logoFileStaticPath  = imageFolderBasePath + "status.png"
+		padding             int
+		LogoBytes           []byte
+	)
+
+	identityImageObjectFromDB, err := multiaccountsDB.GetIdentityImage(keyUID, imageName)
+
+	if err != nil {
+		return nil, err
+	}
+
+	staticLogoFileBytes, _ := images.ImageToBytes(logoFileStaticPath)
+
+	if identityImageObjectFromDB == nil {
+		padding = GetPadding(staticLogoFileBytes)
+		LogoBytes, err = ToLogoImageFromBytes(staticLogoFileBytes, padding)
+	} else {
+		padding = GetPadding(identityImageObjectFromDB.Payload)
+		LogoBytes, err = ToLogoImageFromBytes(identityImageObjectFromDB.Payload, padding)
+	}
+
+	return LogoBytes, err
+}
+
+func GetPadding(imgBytes []byte) int {
+	const (
+		defaultPadding = 20
+	)
+	size, _, err := images.GetImageDimensions(imgBytes)
+	if err != nil {
+		return defaultPadding
+	}
+	return size / 5
+}
+
+func getQrURL(params url.Values, logger *zap.Logger) string {
+	qrURL, ok := params["url"]
+
+	if !ok || len(qrURL) == 0 {
+		logger.Error("[qrops-error] no qr url provided")
+	}
+
+	decodedURL, err := base64.StdEncoding.DecodeString(qrURL[0])
+
+	if err != nil {
+		logger.Error("[qrops-error] unable to decoding string from base64", zap.Error(err))
+	}
+
+	return string(decodedURL)
+}
+
+func getAllowProfileImage(params url.Values, logger *zap.Logger) bool {
+
+	allowProfileImage, ok := params["allowProfileImage"]
+
+	if !ok || len(allowProfileImage) == 0 {
+		// we default to false when this flag was not provided
+		// so someone does not want to allowProfileImage on their QR Image
+		// fine then :)
+		return false
+	}
+
+	LogoOnImage, err := strconv.ParseBool(allowProfileImage[0])
+
+	if err != nil {
+		// maybe for fun someone tries to send non-boolean values to this flag
+		// we also default to false in that case
+		return false
+	}
+
+	// if we reach here its most probably true, but we pass the flag back
+	return LogoOnImage
+
+}
+
+func getLevel(params url.Values, logger *zap.Logger) string {
+	level, ok := params["level"]
+
+	if !ok || len(level) == 0 {
+		// we default to MediumLevel of error correction when the level flag
+		// is not passed.
+		return "2"
+	}
+
+	return level[0]
+}
+
+func getKeyUID(params url.Values, allowProfileImages bool, logger *zap.Logger) string {
+	keyUID, ok := params["keyUid"]
+
+	// if allowProfileImage was also not passed
+	// in that case we don't care and can return an empty string back
+	if !allowProfileImages {
+		return ""
+	}
```

Comment:
If the condition is already known by the consuming code, should this function be responsible for returning an empty value?

Reply:
hmm yeah you are right, I could shift responsibility to the code calling this method to validate if `allowProfileImages` was `false`

Reply:
fixed here. :https://github.com/status-im/status-go/pull/3172/commits/25967f099add2691e53d5558ed8e661d1ab097c4